dotnet/src/SemanticKernel.Abstractions/Contents/FunctionCallResultContent.cs

```diff
@@ -0,0 +1,87 @@
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Represents the result of a function call.
+/// </summary>
+[Experimental("SKEXP0001")]
+public sealed class FunctionCallResultContent : KernelContent
+{
+    /// <summary>
+    /// The function call ID.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public string? Id { get; }
+
+    /// <summary>
+    /// The plugin name.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public string? PluginName { get; }
+
+    /// <summary>
+    /// The function name.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public string? FunctionName { get; }
+
+    /// <summary>
+    /// The result of the function call, the function invocation exception or the custom error message.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public object? Result { get; }
+
+    /// <summary>
+    /// Creates a new instance of the <see cref="FunctionCallResultContent"/> class.
+    /// </summary>
+    /// <param name="functionName">The function name.</param>
+    /// <param name="pluginName">The plugin name.</param>
+    /// <param name="id">The function call ID.</param>
+    /// <param name="result">The function result.</param>
+    [JsonConstructor]
+    public FunctionCallResultContent(string? functionName = null, string? pluginName = null, string? id = null, object? result = null)
+    {
+        this.FunctionName = functionName;
+        this.PluginName = pluginName;
+        this.Id = id;
+        this.Result = result;
+    }
+
+    /// <summary>
+    /// Creates a new instance of the <see cref="FunctionCallRequestContent"/> class.
+    /// </summary>
+    /// <param name="functionCall">The function call.</param>
+    /// <param name="result">The function result.</param>
+    public FunctionCallResultContent(FunctionCallRequestContent functionCall, object? result = null)
+    {
+        this.Id = functionCall.Id;
+        this.PluginName = functionCall.PluginName;
+        this.FunctionName = functionCall.FunctionName;
+        this.Result = result;
+    }
+
+    /// <summary>
+    /// Creates a new instance of the <see cref="FunctionCallRequestContent"/> class.
+    /// </summary>
+    /// <param name="functionCallContent">The function call content.</param>
+    /// <param name="result">The function result.</param>
+    public FunctionCallResultContent(FunctionCallRequestContent functionCallContent, FunctionResult result) :
+        this(functionCallContent, result.Value)
+    {
+        this.InnerContent = result;
+    }
+
+    /// <summary>
+    /// Creates <see cref="ChatMessageContent"/> and adds the current instance of the class to the <see cref="ChatMessageContent.Items"/> collection.
+    /// </summary>
+    /// <returns>The <see cref="ChatMessageContent"/> instance.</returns>
+    public ChatMessageContent ToChatMessage()
```

Comment:
`ToChatMessageContent`?

Reply:
I wanted to keep it short and to omit the "unnecessary" content suffix. Otherwise, I would have to rename the convenience method as well - `FunctionCallRequestContent.GetFunctionCalls` to `FunctionCallRequestContent.GetFunctionCallContents` which would be made it unnecessarily verbose. In any case, I don't mind renaming if really necessary.

Reply:
> `FunctionCallRequestContent.GetFunctionCalls` to `FunctionCallRequestContent.GetFunctionCallContents` which would be made it unnecessarily verbose

I think it's a pattern in other abstractions as well, `IChatCompletionService` as example:
https://github.com/microsoft/semantic-kernel/blob/65275737d66d5c5431656093af541b8667a59720/dotnet/src/SemanticKernel.Abstractions/AI/ChatCompletion/IChatCompletionService.cs#L27

But since it's `Content` class, maybe we can follow different pattern here. I don't think it's really necessary, so feel free to proceed.